src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -89,32 +89,34 @@ public void Add(object key, object value)
 			}
 		}
 
-		public bool ContainsKey(object key) => ContainsKey(key, shouldCheckSystem: true);
-		public bool ContainsKey(object key, bool shouldCheckSystem) => _values.ContainsKey(key) || ContainsKeyMerged(key) || ContainsKeyTheme(key)
+		public bool ContainsKey(object key) => ContainsKey(key, shouldCheckSystem: true, ElementTheme.Default);
+		public bool ContainsKey(object key, bool shouldCheckSystem, ElementTheme elementTheme) => _values.ContainsKey(key) || ContainsKeyMerged(key) || ContainsKeyTheme(key, elementTheme)
 			|| (shouldCheckSystem && !IsSystemDictionary && ResourceResolver.ContainsKeySystem(key));
 
-		public bool TryGetValue(object key, out object value) => TryGetValue(key, out value, shouldCheckSystem: true);
-		public bool TryGetValue(object key, out object value, bool shouldCheckSystem)
+		public bool TryGetValue(object key, out object value) => TryGetValue(key, out value, shouldCheckSystem: true, ElementTheme.Default);
+
+		/// <remarks>Should become internal in the future?</remarks>
```

Comment:
oops! Yes this should be internal, same for the `ContainsKey()` overload

Reply:
Should I change it within this PR?

Reply:
Yes, go ahead

Reply:
Done :-)